apiserver/firewaller/ingressaddresswatcher.go

```diff
@@ -113,6 +124,18 @@ func (w *IngressAddressWatcher) loop() error {
 		return errors.Trace(err)
 	}
 
+	// TODO(wallyworld) - we just want to watch for egress
+	// address changes but right now can only watch for
+	// any model config change.
+	mw := w.backend.WatchForModelConfigChanges()
+	if err := w.catacomb.Add(mw); err != nil {
+		return errors.Trace(err)
+	}
+	// Consume initial event.
+	if _, ok := <-mw.Changes(); !ok {
```

Comment:
The driver of when the ingress address watcher sends its result is the underlying unit(s) entering leaving scope. We call initialise() after consuming the initial model watcher event so the egress-address values are populated prior to entering the for{} loop.